--- a/siod/editline.c
+++ b/siod/editline.c
@@ -176,7 +176,7 @@
 STATIC STATUS h_prev();
 STATIC STATUS h_first();
 STATIC STATUS h_last();
-STATIC int substrcmp(char *text, char *pat, int len);
+STATIC int substrcmp(const char *text, const char *pat, size_t len);
 STATIC ECHAR *search_hist(ECHAR *search, ECHAR *(*move)());
 STATIC STATUS h_search();
 STATIC STATUS fd_char();
@@ -224,10 +224,7 @@
 */
 STATIC ECHAR	*editinput();
 #if	defined(USE_TERMCAP)
-extern char	*getenv();
-extern char	*tgetstr();
-extern int	tgetent();
-extern int	tgetnum();
+#include <term.h>
 #endif	/* defined(USE_TERMCAP) */
 
 /*
@@ -806,7 +803,7 @@
 /*
 **  Return zero if pat appears as a substring in text.
 */
-STATIC int substrcmp(char *text, char *pat, int len)
+STATIC int substrcmp(const char *text, const char *pat, size_t len)
 {
     ECHAR	c;
 
@@ -821,10 +818,10 @@
 STATIC ECHAR *search_hist(ECHAR *search, ECHAR *(*move)())
 {
     static ECHAR	*old_search;
-    int		len;
+    size_t		len;
     int		pos;
-    int		(*match)();
-    char	*pat;
+    int		(*match)(const char *, const char *, size_t);
+    const char	*pat;
 
     /* Save or get remembered search pattern. */
     if (search && *search) {
@@ -841,16 +838,16 @@
     /* Set up pattern-finder. */
     if (*search == '^') {
 	match = strncmp;
-	pat = (char *)(search + 1);
+	pat = (const char *)(search + 1);
     }
     else {
 	match = substrcmp;
-	pat = (char *)search;
+	pat = (const char *)search;
     }
     len = strlen(pat);
 
     for (pos = H.Pos; (*move)() != NULL; )
-	if ((*match)((char *)H.Lines[H.Pos], pat, len) == 0)
+	if ((*match)((const char *)H.Lines[H.Pos], pat, len) == 0)
             return H.Lines[H.Pos];
     H.Pos = pos;
     return NULL;
